src/main/java/file/from/CsvFileReaderImpl.java

```
@@ -0,0 +1,41 @@
+package file.from;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import models.CsvFileReader;
+import models.FruitDao;
+import services.FruitTransaction;
+
+public class CsvFileReaderImpl implements CsvFileReader {
+    private FruitDao fruitDao;
+
+    public CsvFileReaderImpl(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public List<FruitTransaction> read(String filePath) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            boolean isFirstLine = true;
+
+            while ((line = bufferedReader.readLine()) != null) {
+                if (isFirstLine) {
+                    isFirstLine = false;
+                    continue;
+                }
+
+                FruitTransaction fruitTransaction = fruitDao.getFromCsvData(line);
+                transactions.add(fruitTransaction);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("IOException " + e);

```

The error message in the RuntimeException is not sufficiently informative. It should include the file path and a clear description of the error, not just 'IOException ' + e. This violates: 'Don’t ignore exceptions; wrap and rethrow as RuntimeException with informative messages.'